src/utils/ucc_parser.c

```diff
@@ -101,3 +93,41 @@ int ucc_config_names_search(ucc_config_names_array_t *config_names,
 
     return -1;
 }
+
+ucc_status_t ucc_config_allow_list_process(const ucc_config_allow_list_t * list,
+                                           const ucc_config_names_array_t *all,
+                                           ucc_config_allow_list_t *       out)
+{
+    ucc_status_t status;
+    int          i;
+
+    out->mode        = list->mode;
+    out->array.names = NULL;
+
+    if (out->mode == UCC_CONFIG_ALLOW_LIST_ALLOW) {
+        status = ucc_config_names_array_dup(&out->array, &list->array);
+        if (UCC_OK != status) {
+            ucc_error("failed to dup config_names_array");
+            return status;
+        }
+    } else {
```

Comment:
i would rather use switch here, something like
```c
out->mode           = UCC_CONFIG_ALLOW_LIST_ALLOW;
out->array.names = NULL;
swtich (out->mode){
case UCC_CONFIG_ALLOW_LIST_ALLOW:
  ucc_config_names_array_dup(&out->array, &list->array);
case UCC_CONFIG_ALLOW_LIST_ALLOW_ALL:
  ucc_config_names_array_dup(&out->array, &all->array);
case UCC_CONFIG_ALLOW_LIST_NEGATE:
  for (i = 0; i < all->count; i++) {
            if (ucc_config_names_search(&list->array, all->names[i]) < 0)) {
                out->array.names[out->array.count++] = strdup(all->names[i]);
            }
default:
  ucc_error("not supported list mode");
  ucc_assert(0);
}
```

Reply:
looks good (except for mode, see comment above)